src/renderer/widgets/AssetTransactionModal/model/asset-transaction-model.ts

```diff
@@ -0,0 +1,79 @@
+import { combine, createEvent, createStore, restore, sample } from 'effector';
+
+import { includes } from '@shared/lib/utils';
+import { AssetByChains } from '@shared/core';
+import { portfolioModel } from '@features/assets';
+import { ModalType, Step } from '../lib/types';
+
+const flowClosed = createEvent();
+
+const stepChanged = createEvent<Step>();
+const modalTypeChanged = createEvent<ModalType>();
+const queryChanged = createEvent<string>();
+
+const $selectedAsset = createStore<AssetByChains | null>(null).reset(flowClosed);
+const $modalType = restore<ModalType | null>(modalTypeChanged, null).reset(flowClosed);
+const $step = restore<Step>(stepChanged, Step.NONE).reset(flowClosed);
+const $query = restore<string>(queryChanged, '').reset(flowClosed);
+
+const $assetWithChains = combine(
+  {
+    selectedAsset: $selectedAsset,
+    query: $query,
+  },
+  ({ selectedAsset, query }) => {
+    if (!query || !selectedAsset) return selectedAsset;
+
+    const filteredChains = selectedAsset.chains.filter((chain) => {
+      const hasSymbol = includes(chain.assetSymbol, query);
+      const hasChainName = includes(chain.name, query);
+
+      return hasSymbol || hasChainName;
+    });
+
+    return { ...selectedAsset, chains: filteredChains };
+  },
+  { skipVoid: false },
```

Comment:
Seems to me that `$assetWithChains` returns `null | {} | AssetByChains` but not `undefined`
thus `skipVoid` is redundant